repro.in

```diff
@@ -221,11 +255,13 @@ __END__
 function init_chroot(){
     mkdir -p "$BUILDDIRECTORY"
 
-    # Prepare root chroot
+    # Always lock first. Otherwise we might end up...
+    # - doing the same thing again - if using test/lock/mkdir
+    # - with empty directory in the follow-up lock - if using test/mkdir/lock
+    lock 9 "$BUILDDIRECTORY"/root.lock
```

Comment:
We run https://reproducible.archlinux.org/ with 5 workers. If one worker is in the middle of building a package there would be 4 workers blocked at this part as it can't acquire the lock. This time could be better spent preparing the next keyring, or fetching packages to the cache.

Reply:
For this and the other locking examples - I'm thinking that we might want to add some tests.

In particular add a few strategical placeholders through the code, then `make check` will:
 - replace individual placeholders with a stall like `sleep 10000` and save them as different variants
 - for each variant stub out systemd-nspawn perhaps?
 - run the variants concurrently

Reply:
Having a better way to test this thing would be great, but I suspect complicated. My way to test things has mainly been running 2-3 repros concurrently and see what blocks where and how.